### PR TITLE
Add Docker support

### PR DESCRIPTION
**Pull request by gnuvince**
This requires testing by other people.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+     && ninja -C . install
+
+FROM debian:9.4-slim
+
+RUN apt-get update                                \
+    && apt-get install -y --no-install-recommends \
+        ca-certificates                           \
+        libbsd0                                   \
+        libdaemon0                                \
+        libmicrohttpd12                           \
```
Comment (RAttab) [inline Dockerfile]: I wonder if they have the epoll support flipped on by default in this package but I think the answer is probably yes.

Comment (RAttab) [reply Dockerfile]: doesn't matter a whole lot either way though.

**Comment by RAttab** [comment]
Looks good for the most part but would need to try it out real quick.

**Comment by gnuvince** [comment]
To test, you can start the container (instructions are in README) and then compile and run `test/example.c`; I had to add a `fgets()` call in there to slow down the client.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,39 @@
+FROM debian:9.4 AS build
+
+RUN apt-get update                                \
+    && apt-get install -y --no-install-recommends \
+        build-essential                           \
+        ca-certificates                           \
+        cmake                                     \
+        git                                       \
```
Comment (marina101) [inline Dockerfile]: why is git in here?

**Comment by marina101** [comment]
Mac update: now this builds/tests run on mac. We realized we can't actually _run_ optics on mac though, so compiling example.c breaks. But this makes sense, given that optics is linux only. It is still helpful for being able to develop/test on a mac for someone who doesn't have a vm.

File: Dockerfile (hunk 2 above)
Comment (gnuvince) [reply Dockerfile]: Removed!

**Comment by marina101** [comment]
I'd be interested in working on a mac patch. Maybe I could build some sort of adapter so that you could keep the main interface consistent? Anyways, we should sit down and discuss this at some point

**Pull request merged**